File: corpus/typescript/fixture_08.ts
type CartItem8 = { price: number; qty?: number };
export function calculateDiscount8(cartItems: CartItem8[], taxRate: number): number {
  const totalAmount = cartItems.reduce((sum, item) => sum + item.price * (item.qty ?? 1), 0);
  return totalAmount > 80 ? totalAmount * (1 + taxRate) : totalAmount;
}

export class PriceBook8 {
  lookupPrice(sku: string): number { return sku.length + 8; }
}

File: corpus/go/fixture_08.go
package fixture8

type PriceBook8 struct {
    Multiplier float64
}

func CalculateDiscount8(cartItems []float64, taxRate float64) float64 {
    totalAmount := 0.0
    for _, price := range cartItems {
        totalAmount += price
    }
    if totalAmount > 80.0 {
        return totalAmount * (1.0 + taxRate)
    }
    return totalAmount
}
